src/kernel/diffusive/diffusive.f90

```diff
@@ -1600,16 +1668,26 @@ function rtsafe(i, j, Q_cur, Q_ds, z_cur, z_ds, y_ds)
         dxold  = dxx
         dxx    = 0.50 * (xh - xl)
         rtsafe = xl + dxx
-        if (xl == rtsafe) return   ! change in root is negligible.
+         !write(301, *) "iter:", i, j, iter, rtsafe
+        if (xl == rtsafe) then 
+         !write(301, *) "fnal:", i, j, iter, rtsafe
+         return   ! change in root is negligible.
+        endif 
       else                         ! newton step acceptable. take it.
         dxold  = dxx
         dxx    = f / df
         temp   = rtsafe
         rtsafe = rtsafe - dxx
-        if (temp == rtsafe) return
+        if (temp == rtsafe) then
```

Comment:
remove this whole block since it is a `noop` now anyways.

Reply:
removed. thanks.